src/components/organisms/AdminVenueView/components/EventsView/CsvImportModal.tsx

```diff
@@ -0,0 +1,219 @@
+    if (!parsedData) {
+      return;
+    }
+
+    setImportResult(undefined);
+    const result = await importEventsBatch(transformImportToEvents(parsedData));
+    setImportResult(result);
+  }, [parsedData]);
+
+  const hasParseError = parsedErrors && parsedErrors.length > 0;
+  const hasImportError = !!importError;
+
+  const dialogClassName = classNames({
+    "CsvImportModal CsvImportModal__dialog": true,
+    "CsvImportModal--with-data": parsedData,
+    "CsvImportModal--with-error": hasParseError || hasImportError,
+  });
+
+  return (
+    <Modal
+      centered
+      dialogClassName={dialogClassName}
+      onHide={onHide}
+      show={show}
+    >
+      <Modal.Header className="CsvImportModal__header">
+        <Modal.Title className="CsvImportModal__title">
+          Import from CSV
+        </Modal.Title>
+      </Modal.Header>
+      <Modal.Body className="CsvImportModal__body">
+        {importResult && (
+          <div className="CsvImportModal__result">
+            <div className="CsvImportModal__result-count">
+              The number of new events is {importResult.updateCount}.
+            </div>
+            {importResult.errors.length > 0 && (
+              <div className="CsvImportModal__result-errors">
+                Some events weren&apos;t addded due to invalid data. Please
+                check lines:
+                {" " +
+                  importResult.errors
+                    .map(({ index, reason }) => `#${index + 1} (${reason})`)
+                    .join(", ")}
+                .
+              </div>
+            )}
+          </div>
+        )}
+        {hasImportError && (
+          <div className="CsvImportModal__error CsvImportModal__error--import">
+            <span className="CsvImportModal__error-static">
+              There has been an error importing the data. Error:
+            </span>
+            <br />
+            <span className="CsvImportModal__error-generated">
+              {"" + importError?.message}
+            </span>
+          </div>
+        )}
+        {hasParseError && parsedErrors?.length === 1 && (
+          <div className="CsvImportModal__error CsvImportModal__error--single">
+            <span className="CsvImportModal__error-static">
+              There has been an error parsing the CSV file. Please make sure
+              everything is OK before proceeding. Error:
+            </span>
+            <br />
+            <span className="CsvImportModal__error-generated">
+              {parsedErrors[0].message}
+            </span>
+          </div>
+        )}
+        {hasParseError && parsedErrors && parsedErrors?.length > 1 && (
+          <div className="CsvImportModal__error CsvImportModal__error--multiple">
+            <span className="CsvImportModal__error-static">
+              There have been multiple errors parsing the CSV file. Please make
+              sure everything is OK before proceeding. Errors:
+            </span>
```

Comment:
I think you should add the `.length` check here and change only the `span` content. The map a few lines below should be working fine with a single error.

Reply:
Made the code more readable at the expense of the messages readability @ 75433ea53318970aadf7fafe4e4d2087dba2ef41